src/screens/timeStamping/TimeStampingProfile/index.tsx

```diff
@@ -18,44 +19,66 @@ const renderEvent = (event: EventType) => (
 );
 
 const TimeStampingProfile = () => {
-  const [currentDate, setCurrentDate] = useState<Date>(new Date());
+  const [displayedDate, setDisplayedDate] = useState<Date>(new Date());
   const [events, setEvents] = useState<EventType[]>([]);
-
+  const [isAppBackgrounded, setIsAppBackgrounded] = useState<boolean>(false);
   const { loggedUser } = useContext(AuthContext);
 
   useEffect(() => {
-    const interval = setInterval(() => { setCurrentDate(new Date()); }, 60000);
+    const interval = setInterval(() => { setDisplayedDate(new Date()); }, 60000);
+
     return () => clearInterval(interval);
   }, []);
 
-  const fetchInterventions = useCallback(async () => {
-    try {
-      if (!loggedUser || !loggedUser._id) return;
-      const params = {
-        auxiliary: loggedUser._id,
-        startDate: new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate(), 0, 0, 0, 0),
-        endDate: new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDate(), 23, 59, 59, 999),
-        type: INTERVENTION,
+  useFocusEffect(
+    useCallback(() => {
+      let isActive = true;
+      let appState = AppState.currentState;
+
+      const handleBackground = (nextAppState: AppStateStatus) => {
+        setIsAppBackgrounded(false);
+        if ((appState === 'background' || appState === 'inactive') && nextAppState === ACTIVE_STATE) {
+          setIsAppBackgrounded(true);
+        }
+        appState = nextAppState;
       };
-      const fetchedEvents = await Events.events(params);
 
-      setEvents(fetchedEvents);
-    } catch (e) {
-      console.error(e);
-    }
-  }, [loggedUser, currentDate]);
+      AppState.addEventListener('change', handleBackground);
 
-  useEffect(() => {
-    if (loggedUser?._id) fetchInterventions();
-  }, [loggedUser, fetchInterventions]);
+      const fetchInterventions = async () => {
+        const today = new Date();
+        try {
+          if (!loggedUser || !loggedUser._id) return;
+          const params = {
+            auxiliary: loggedUser._id,
+            startDate: new Date(today.getFullYear(), today.getMonth(), today.getDate(), 0, 0, 0, 0),
+            endDate: new Date(today.getFullYear(), today.getMonth(), today.getDate(), 23, 59, 59, 999),
+            type: INTERVENTION,
+          };
+          const fetchedEvents = await Events.events(params);
+
+          if (isActive || isAppBackgrounded) setEvents(fetchedEvents);
```

Comment:
En quoi a-t-on besoin de `isAppBackgrounded` ici ?

Reply:
Non on en a pas besoin en effet! Je l'avais ajouter pour que es-lint ne mette pas d'erreur dans le tableau de dépendances. (sinon es-lint renvoyait l'erreur `React Hook useCallback has an unnecessary dependency: 'appState'. Either exclude it or remove the dependency array.`)